scenario/node/lib/scenario_steps.js

```diff
@@ -244,13 +252,46 @@ When('I invoke the transaction', async () => {
   }
 });
 
+When('I invoke the transaction which I expect to fail', async () => {
+  const proposal = this.contract.newProposal(this.txn.name, this.txn.options);
+  try {
+    if (this.txn.type === 'evaluate') {
+      this.txn.result = await proposal.evaluate();
+    } else if (this.txn.type === 'submit') {
+      const transaction = await proposal.endorse();
+      this.txn.result = await transaction.submit();
+    } else {
+      //throw new Error(`Unknown transaction type: ${this.txn.type}`);
```

Comment:
The tests are broken if we get here so we should throw

Reply:
Actually, most of this function is a duplicate of the implementation of `When('I invoke the transaction')`. Perhaps we could factor out some common code

Reply:
> The tests are broken if we get here so we should throw

It will throw, but admittedly with a misleading message - I'll fix that.